services/steam/steam.service.js

```diff
@@ -0,0 +1,395 @@
+'use strict'
+
+const Joi = require('joi')
+const BaseJsonService = require('../base-json')
+const { metric, formatDate } = require('../../lib/text-formatters')
+const { age: ageColor } = require('../../lib/color-formatters')
+const prettyBytes = require('pretty-bytes')
+
+const steamCollectionSchema = Joi.object({
+  response: Joi.object()
+    .keys({
+      result: Joi.number()
+        .integer()
+        .min(1)
+        .max(1)
+        .required(),
+      resultcount: Joi.number()
+        .integer()
+        .min(0)
+        .max(1)
+        .required(),
+      collectiondetails: Joi.array()
+        .items(
+          Joi.object({
+            publishedfileid: Joi.string(),
+            result: Joi.number().integer(),
+            children: Joi.array(),
```

Comment:
`result` and `children` should be `.required()` here. Any property we're going to reference without a `try`/`catch` in the `handle()` should be `.required()` in the validator.
Does `publishedfileid` actually get used? If not, we can exclude it from the validator.